src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -233,6 +234,8 @@ public static int getVersionOrdinal(int version)
         return ordinal;
     }
 
+    public final static boolean NON_GRACEFUL_SHUTDOWN = Boolean.getBoolean("cassandra.test.messagingService.nonGracefulShutdown");
```

Comment:
@josh-mckenzie great catch, yes, it definitely should be there. If we managed to have a second reviewer for CASSANDRA-17797, we would merge that and this would fail the build :)

Reply:
Quite interesting question. This is something to be used only in tests. Perhaps something overly contrived but maybe test properties should require something like `cassandra.test` property to be set in order to make them get assigned with a non-default value? (I mean, set `cassandra.test` to unlock those properties). Perhaps something for CASSANDRA-17797 @smiklosovic

Reply:
The patch we work on is gathering all properties, irrelevant what they are for, in one place. I understand these are just test properties but having them somewhere separately, for example, in test source, would mean that we would start to reference test class in the prod code. Right? Not good. Having test code referencing prod code where these test properties are is better.

I am not sure how to replay to your idea about `cassandra.test`. I think that in general it is good idea to have them with that prefix, yes. Otherwise we are not completely sure, at the first glance, if there is not any comment, what that property is actually used for / upon.

In the long run, we would like to have all test properties with such prefix but the first task is to manage the current "chaos" with having them scattered across all the source code. As of now, nobody has a definitive answer what all properties we use ....

Reply:
I just meant to prevent setting those properties to some non-default values in production, that is, you can only override the defaults if some additional property like, `-Dcassandra.test=true` is set

Reply:
ah right ...  @Mmuzaf this is an interesting point to take into account